plugin/src/main/kotlin/com/automattic/android/applystyle/ApplyStylePlugin.kt

```diff
@@ -0,0 +1,167 @@
+/*
+ * This Kotlin source file was generated by the Gradle 'init' task.
+ */
+package com.automattic.android.applystyle
+
+import org.gradle.api.Project
+import org.gradle.api.Plugin
+
+import io.gitlab.arturbosch.detekt.DetektPlugin
+import io.gitlab.arturbosch.detekt.extensions.DetektExtension
+import org.gradle.api.plugins.quality.CodeQualityExtension
+import java.io.File
+import java.io.IOException
+import java.nio.file.Path
+
+/**
+ * A custom plugin that standardizes the styles across various projects.
+ *
+ * For now the plugin deals only with the configuration of Detekt.
+ */
+class ApplyStylePlugin: Plugin<Project> {
+    override fun apply(project: Project) {
+        project.tasks.register("extractDetektConf") { task ->
+            task.doLast {
+                val detektExtension = project.getDetektExtension() as DetektExtension
+                val file = detektExtension.config.singleFile ?: path(project.buildDir,"detekt.yml")
+                project.extractDetektConf(file, true)
+                println("Wrote detekt configuration file to $file")
+            }
+        }
+
+        project.tasks.register("greeting") { task ->
+            task.doLast {
+                println("Hello from plugin '$PLUGIN_NAME'")
+            }
+        }
```

Comment:
We probably don't need this task anymore.

Reply:
It is referenced by the exiting functional tests. That's why it is still there.